volatility3/framework/plugins/windows/devicetree.py

```diff
@@ -0,0 +1,145 @@
+# This file is Copyright 2022 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+
+from typing import Iterator, List, Tuple
+
+from volatility3.framework import constants, renderers, exceptions, interfaces
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins.windows import driverscan
+
+DEVICE_CODES = {
+    0x00000027 : "FILE_DEVICE_8042_PORT",
+    0x00000032 : "FILE_DEVICE_ACPI",
+    0x00000029 : "FILE_DEVICE_BATTERY",
+    0x00000001 : "FILE_DEVICE_BEEP",
```

Comment:
In general, I think it is appropriate for a structured data blueprint or a specific value enumeration to be a JSON-type Symbol. That way, developers can focus on code flow and logic.
However, in this case, I only need the Device Code, so I inserted it into the code, but it doesn't seem to be my favorite result. If you have any good ideas or cases, please let me know! 🤔

Reply:
I will solve this conversations with the under review!

Reply:
Yep, I don't have any good ideas, it would be quite complex to put it in a JSON ISF file just for the sake of an enum lookup, so as I say, in this instance I think you've made the right choice.  5:)